packages/twenty-server/src/database/commands/upgrade-version/0-24/0-24-migrate-email-fields-to-emails.command.ts

```diff
@@ -0,0 +1,305 @@
+
+import chalk from 'chalk';
+import { Command, Option } from 'nest-commander';
+import { QueryRunner, Repository } from 'typeorm';
+
+import { ActiveWorkspacesCommandRunner } from 'src/database/commands/active-workspaces.command';
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { DataSourceEntity } from 'src/engine/metadata-modules/data-source/data-source.entity';
+import { DataSourceService } from 'src/engine/metadata-modules/data-source/data-source.service';
+import { CreateFieldInput } from 'src/engine/metadata-modules/field-metadata/dtos/create-field.input';
+import {
+  FieldMetadataEntity,
+  FieldMetadataType,
+} from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { FieldMetadataService } from 'src/engine/metadata-modules/field-metadata/field-metadata.service';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { computeTableName } from 'src/engine/utils/compute-table-name.util';
+import { ViewService } from 'src/modules/view/services/view.service';
+import { ViewFieldWorkspaceEntity } from 'src/modules/view/standard-objects/view-field.workspace-entity';
+
+interface MigrateEmailFieldsToEmailsCommandOptions {
+  workspaceId?: string;
+}
+
+@Command({
+  name: 'upgrade-0.24:migrate-email-fields-to-emails',
+  description: 'Migrating fields of deprecated type EMAIL to type EMAILS',
+})
+export class MigrateEmailFieldsToEmailsCommand extends ActiveWorkspacesCommandRunner {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    protected readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+    private readonly fieldMetadataService: FieldMetadataService,
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly viewService: ViewService,
+  ) {
+    super(workspaceRepository);
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description:
+      'workspace id. Command runs on all active workspaces if not provided',
+    required: false,
+  })
+  async executeActiveWorkspacesCommand(
+    _passedParam: string[],
+    options: MigrateEmailFieldsToEmailsCommandOptions,
+    workspaceIds: string[],
+  ): Promise<void> {
+    this.logger.log(
+      'Running command to migrate email type fields to emails type',
+    );
+    const _workspaceIds = options.workspaceId
```

Comment:
workspaceIds already contains the correct set of ids (options if present, all active workspaces if not present), you can use it directly. This instantiation is actually not correct because if we use "-w" options in the command we want to apply only this workspaceId and not combine with the others like you did here
```typescript
[...workspaceIds, options.workspaceId]
```